packages/flutter/lib/src/services/autofill.dart

```diff
@@ -799,13 +799,36 @@ mixin AutofillScopeMixin implements AutofillScope {
       !autofillClients.any((AutofillClient client) => client.textInputConfiguration.autofillConfiguration == null),
       'Every client in AutofillScope.autofillClients must enable autofill',
     );
-    return TextInput.attach(
-      trigger,
-      _AutofillScopeTextInputConfiguration(
-        allConfigurations: autofillClients
-          .map((AutofillClient client) => client.textInputConfiguration),
-        currentClientConfiguration: configuration,
-      ),
+
+    TextInputConfiguration inputConfiguration;
+    switch (defaultTargetPlatform) {
+      case TargetPlatform.iOS:
+        bool isPassword(AutofillClient client) {
+          final Iterable<String> hints = client.textInputConfiguration.autofillConfiguration?.autofillHints;
+          return client.textInputConfiguration.obscureText
+            || (hints?.contains(AutofillHints.password) ?? false)
+            || (hints?.contains(AutofillHints.newPassword) ?? false);
```

Comment:
Question: why are we adding this check in the framework, can we handle it on the iOS engine side?

The reason that I'm concerned is, platform is does not necessarily mean that this is a flutter app running as an iOS mobile app. It can also be a Flutter Web app running on iOS Safari. Since web also return TargetPlatform.iOS, in this case, this branch won't work correctly on browsers on iOS, running Flutter for Web. The code is here: https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/foundation/_platform_web.dart#L20

Reply:
I'll check if `kIsWeb`.